### PR TITLE
Auth and unauth guards

### PR DESCRIPTION
**Pull request by joacomiguens**
## Summary

- Reestructure app with two main screens auth and unauth.

- Create Route Guards.

## Screenshot

guards

## Trello Card

You can find it [here](https://trello.com/c/oYNb8cIT/15-auth-guard-unauth-guard)

### CODE REVIEW

File: wolox-books/src/app/screens/auth/screens/book-list/book-list.component.html [hunk 1]
```diff
@@ -1,3 +1,4 @@
+<app-navbar></app-navbar>
```
Comment (ebedoyaalzate) [inline wolox-books/src/app/screens/auth/screens/book-list/book-list.component.html]: Is necessary add navbar here?? you already put navbar component in the parent

File: wolox-books/src/app/screens/unauth/unauth.component.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-unauth',
+  templateUrl: './unauth.component.html',
+  styleUrls: ['./unauth.component.scss'],
+})
+export class UnAuthComponent implements OnInit {
+  constructor() {}
```
Comment (ebedoyaalzate) [inline wolox-books/src/app/screens/unauth/unauth.component.ts]: add empty line

**Review by ebedoyaalzate — CHANGES_REQUESTED** [review]

File: wolox-books/src/app/screens/auth/screens/book-list/book-list.component.html (hunk 1 above)
Comment (joacomiguens) [reply wolox-books/src/app/screens/auth/screens/book-list/book-list.component.html]: Sorry, fixed 😃

**Review by ebedoyaalzate — APPROVED** [review]

File: wolox-books/src/app/guards/auth.guard.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  constructor(private userService: UserService, private router: Router) {}
+
+  canActivate(): boolean {
+    const hasToken = this.userService.hasToken();
+    if (hasToken) {
+      return true;
+    } else {
+      this.router.navigate(['/login']);
+      return false;
+    }
```
Comment (mcallegari10) [inline wolox-books/src/app/guards/auth.guard.ts]: Mmm, maybe this is cleaner
```suggestion
    if (!hasToken) this.router.navigate(['/login']);
    return hasToken;
```

File: wolox-books/src/app/guards/unauth.guard.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UnAuthGuard implements CanActivate {
+  constructor(private userService: UserService, private router: Router) {}
+
+  canActivate(): boolean {
+    const hasToken = this.userService.hasToken();
+    if (hasToken) {
```
Comment (mcallegari10) [inline wolox-books/src/app/guards/unauth.guard.ts]: Same here

**Review by mcallegari10 — APPROVED** [review]

**Pull request merged**